src/Tools/Pagination/LimitSubqueryOutputWalker.php

```diff
@@ -158,11 +164,33 @@ private function rebuildOrderByForRowNumber(SelectStatement $AST): void
      */
     public function walkSelectStatement(SelectStatement $AST)
     {
+        $sqlFinalizer = $this->getFinalizer($AST);
```

Comment:
Its unfortunate this API only works for SELECt statements. First, the walkSelectStatement method does not work like any other method anymore, breaking the implicit contract of what these methods do in the SqlWalker. Shouldn't we refactor the APIs completly in a way thats similar to executors? I.e. SqlOutputWalker has a method `geSqlFinalizer($AST)` similar to `public function getExecutor($AST)`?

Then for all non select statements it could return a simple implementation.

But looking at SqlExecutors, why not stay with this abstraction? Its almost capable of performing the last steps in `AbstractSqlExecutor::execute`. We could introduce `AbstractSqlExecutor::finalizeAndExecute(Query $query, ...)` - this would have access to the necessary paraemters to perform the last two finalizing steps (limiting, locking).